client/src/components/StoreList.tsx

```diff
@@ -1,65 +1,136 @@
-    setUserCoords([position.coords.latitude, position.coords.longitude]);
-  }
+  // Dummy map attachment
+  const map = new google.maps.Map(document.getElementById("mapPlaceholder"));
+  // Declare our Google Map/Places API service handler
+  const service = new google.maps.places.PlacesService(map);
 
-  function errorbackLoc(error: any) {
+  // Error callback for location API call
+  const errorbackLoc = (error: any) => {
     alert(error);
-  }
+  };
 
-  // fetch user location
-  grabLoc();
+  // Grab the location of device
+  const grabLoc = () => {
+    // Use microapps API if we are in an iframe
+    if (window.location !== window.parent.location) {
+      setIsLoading(true);
+      microapps
+        .getCurrentLocation()
+        .then((loc: any) => {
+          const position = {
+            coords: {
+              latitude: loc["latitude"],
+              longitude: loc["longitude"],
+            },
+          };
+          fetchStores(position);
+        })
+        .catch((err: any) => errorbackLoc(err));
+    }
+    // Otherwise, use browser navigator
+    else if (navigator.geolocation) {
+      setIsLoading(true);
+      navigator.geolocation.getCurrentPosition(fetchStores, errorbackLoc);
+    }
+  };
 
   // fetch list of users
-  const fetchStores = async () => {
+  const fetchStores = async (position: {
+    coords: {
+      latitude: number;
+      longitude: number;
+    };
+  }) => {
     // Update location
     grabLoc(); //blocking? race on userCoords update?
     const data = {
       distance: 10000,
-      latitude: userCoords[0],
-      longitude: userCoords[1],
+      latitude: position.coords.latitude,
+      longitude: position.coords.longitude,
     };
     const stores = await fetchJson(data, "/api/stores");
     setStoreList(stores);
+    setIsLoading(false);
   };
 
-  function LocationTrack(props: any) {
-    return (
-      <h3>
-        [{userCoords[0]},{userCoords[1]}]
-      </h3>
-    );
-  }
+  // Request PlacesAPI for nearby locations
+  const getNearbyPlaces = () => {
+    const curLoc = new google.maps.LatLng(userCoords[0], userCoords[1]);
+    const request = {
+      location: curLoc,
+      radius: "5000",
+      type: ["restaurant"],
+    };
+    setIsLoading(true);
+    service.nearbySearch(request, getNearbyPlacesCallback);
+  };
+
+  // Digest PlacesAPI results
+  const getNearbyPlacesCallback = (places: any, status: any) => {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+      setNearbyPlaces(places);
+    } else {
+      alert("PlacesAPI failure to return");
+    }
+    setIsLoading(false);
+  };
+
+  // Get user identity details
+  const loginUser = async () => {
+    // Only run in iframe (on app)
+    if (window.location !== window.parent.location) {
+      const request = { nonce: "Don't Hack me please" };
+      microapps
+        .getIdentity(request)
+        .then((response: any) => {
+          const decoded = JSON.parse(atob(response.split(".")[1]));
```

Comment:
Add a comment to explain what you are doing here. Otherwise, complicated logic like this may be confusing for other developer (even you after some while)

Reply:
Code sample taken from Spot API identity call. Documentation sparse, looked into response structure to determine reasoning behind this line [commented].